### PR TITLE
refactor(client-debugger): Update library to use global singleton for managing the root devtools object

### PR DESCRIPTION
**Pull request by Josmithr**
Updates the devtools library to not require consumers to explicitly dispose of their devtools object during app teardown. This has a couple of major benefits:
1. Requiring consumers to dispose of the object puts substantial burden on consumers to keep track of the devtools object in all relevant places, which may or may not be reasonable for a (presumably) dev-build-only entity.
2. We can ensure teardown occurs on page close / refresh, which helps ensure the Chrome extension can clean up stale data.

This PR also updates the package's `packageDocumentation` comment and README overview to reflect the new flow (and cleans up a significant amount of outdated info).

See #15312 for an example of how this improves usage in application flow.

### CODE REVIEW

File: packages/tools/client-debugger/client-debugger/.eslintrc.js [hunk 1]
```diff
@@ -18,12 +18,15 @@ module.exports = {
 	overrides: [
 		{
 			// Overrides for test files
-			files: ["*.spec.ts", "*.test.ts", "src/test/**"],
+			files: ["*.test.ts", "src/test/**"],
```
Comment (Josmithr) [inline packages/tools/client-debugger/client-debugger/.eslintrc.js:21]:
```suggestion
			files: ["*.spec.ts", "*.test.ts", "src/test/**"],
```

File: packages/tools/client-debugger/client-debugger/src/index.ts [hunk 2]
```diff
@@ -4,41 +4,15 @@
...
+ * Contains an API for initializing developer tooling alongside the Fluid Framework.
  *
- * Debugger instances are created per `Container` instance and are bound to the `globalThis` context
- * to be accessible to related tools.
+ * This tooling is built for use alongside our Chromium browser extension.
+ * It listens for incoming messages from the corresponding extension, and posts direct responses as well as automatic
+ * updates for Fluid state changes.
  *
- * - See the package README for more details on related tools that are enabled via these debugger instances.
+ * Individual {@link IFluidContainer | Fluid Containers} can be registered to generate Container-level stats.
```
Comment (Josmithr) [inline packages/tools/client-debugger/client-debugger/src/index.ts:13]:
```suggestion
 * Individual {@link @fluidframework/container-definitions#IContainer | Fluid Containers} can be registered to generate Container-level stats.
```

**Review by alexvy86 — APPROVED** [review]

File: packages/tools/client-debugger/client-debugger/README.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # @fluid-tools/client-debugger
 
-The Fluid Client Debugger library contains a simple API for initializing debug sessions for recording and propogating information about a given Fluid [Container][] and its [Audience][].
+This library contains a library of developer tools for use alongside the Fluid Framework.
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply packages/tools/client-debugger/client-debugger/README.md:3]: Haha, yeah that is awkward phrasing isn't it? What do you think of the following instead:

```suggestion
This library contains developer tools for use alongside the Fluid Framework.
```

[the remark this answers is not in the document]
Comment (alexvy86) [reply packages/tools/client-debugger/client-debugger/README.md:3]: Yep, that works 🚀

**Pull request merged**